From 80e64a8f6b830bef5c4d1abd4a2af2776b254760 Mon Sep 17 00:00:00 2001
From: CarlJay <1334261611@qq.com>
Date: Mon, 19 Apr 2021 20:40:56 +0800
Subject: [PATCH] Optimize code in phase1.hpp

	Util::IntTo16Bytes(to_write, entry);
	memcpy(&(right_writer_buf[i * entry_size_bytes]), to_write, 16);
instead of:
	Util::IntTo16Bytes(&right_writer_buf[i * entry_size_bytes], entry);
Delete memcpy
---
 src/phase1.hpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/phase1.hpp b/src/phase1.hpp
index edace623..fe481b1e 100644
--- a/src/phase1.hpp
+++ b/src/phase1.hpp
@@ -558,13 +558,11 @@ void* F1thread(int const index, uint8_t const k, const uint8_t* id, std::mutex*
         // to increase CPU efficency.
         f1.CalculateBuckets(x, loopcount, f1_entries.get());
         for (uint32_t i = 0; i < loopcount; i++) {
-            uint8_t to_write[16];
             uint128_t entry;
 
             entry = (uint128_t)f1_entries[i] << (128 - kExtraBits - k);
             entry |= (uint128_t)x << (128 - kExtraBits - 2 * k);
-            Util::IntTo16Bytes(to_write, entry);
-            memcpy(&(right_writer_buf[i * entry_size_bytes]), to_write, 16);
+            Util::IntTo16Bytes(&right_writer_buf[i * entry_size_bytes], entry);
             right_writer_count++;
             x++;
         }
